src/main/java/com/fasterxml/jackson/perf/json/JsonArbitraryFieldNameBenchmark.java

```diff
@@ -0,0 +1,151 @@
+            @Override
+            JsonParser create(JsonFactory factory, Supplier<String> jsonSupplier) throws IOException {
+                return factory.createParser(new ByteArrayInputStream(jsonSupplier.get().getBytes(StandardCharsets.UTF_8)));
+            }
+        },
+        READER() {
+            @Override
+            JsonParser create(JsonFactory factory, Supplier<String> jsonSupplier) throws IOException {
+                // Instead of using 'new StringReader(jsonSupplier.get())', we construct an InputStreamReader
+                // to more closely match overhead of INPUT_STREAM for comparison.
+                return factory.createParser(new InputStreamReader(
+                        new ByteArrayInputStream(jsonSupplier.get().getBytes(StandardCharsets.UTF_8)),
+                        StandardCharsets.UTF_8));
+            }
+        };
+
+        abstract JsonParser create(JsonFactory factory, Supplier<String> jsonSupplier) throws IOException;
+    }
+
+    public enum InputShape {
+        RANDOM_KEY_MAP(
+                new TypeReference<Map<String, Boolean>>() {},
+                () -> "{\"" + ThreadLocalRandom.current().nextInt() + "\":true}"),
+        BEAN_WITH_RANDOM_KEY_MAP(
+                new TypeReference<SimpleClass>() {},
+                () -> "{\"fieldWithMap\":{\"" + ThreadLocalRandom.current().nextInt()
+                        + "\":true},\"stringOne\":\"a\",\"stringTwo\":\"a\",\"stringThree\":\"a\"}");
+
+        private final TypeReference<?> typereference;
+        private final Supplier<String> jsonSupplier;
+        InputShape(TypeReference<?> typereference, Supplier<String> jsonSupplier) {
+            this.typereference = typereference;
+            this.jsonSupplier = jsonSupplier;
+        }
+    }
+
+    @Param
+    public InputShape shape;
+
+    @Param
+    public InputType type;
+
+    @Param
+    public FactoryMode mode;
+
+    private JsonFactory factory;
+    private ObjectReader reader;
+
+    @Setup
+    public void setup() {
+        factory = mode.apply(new JsonFactory());
+        ObjectMapper mapper = new ObjectMapper(factory)
+                // Use FAIL_ON_UNKNOWN_PROPERTIES to ensure the benchmark inputs are valid
+                .enable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        reader = mapper.readerFor(shape.typereference);
+    }
+
+    @Benchmark
+    public Object parse() throws IOException {
+        try (JsonParser parser = type.create(factory, shape.jsonSupplier)) {
+            return reader.readValue(parser);
+        }
+    }
+
+    /**
+     * This type primarily exists to wrap a map, but has additional
+     * fields to cover a mix of reused and arbitrary json keys.
+     */
+    public static final class SimpleClass {
+        public Map<String, Boolean> fieldWithMap;
+
+        public String stringOne;
+
+        public String stringTwo;
+
+        public String stringThree;
+
+        @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
```

Comment:
Instead of using Creators, maybe simple setters/getters; this is faster than using Constructors.

Reply:
I considered avoiding databinding altogether, but it's a convenient way to consume the data with some level of validation. Happy to update to getters and setters (or perhaps public property fields?).

Reply:
Yeah not a huge deal by any measure, either works fine.